WSUCptSCapstone-Fall2022Spring2023/psd-gamifiedapp
# Sprint3/main levels

Closes #13 

Demo Video Link: https://youtu.be/PGIrg8Q7YOM

Overview:

This pull requests uses the existed level framework to implement the 5 committed to main levels covering concepts of statements, conditionals, loops, and lists in Python. The main levels each feature multiple phases and address all important concepts involved in the main topics of each level. It also makes some ancillary changes to the level player UI and code to facilitate the needs of actual in-practice levels.

Details:

This PR changes the following code files:

- IronPythonContainer.cs: Added additional utilities for the Python test files, made sure that partial code simulation worked properly.
- OutputController.cs: Added support for multi-phase scripts and added better failure message support. Also tied it to an onclick event so that one can click to advanced through scripts in the output window.
- This PR also adds _many_ text files for both lesson Python and lessons scripts. Due to the number of files they are only referenced here in one bullet point.

Testing:

The functionality for this pull request was tested using the following manual FT procedure.

- Start game
- For each level in the level list:
- Start level
- Click through all phase dialogue
- Enter expected failure
- Confirm failure message
- Enter solution
- Confirm success and proceed to next phase

In the current build all main levels can be played through in this way with expected failures and successes triggering as anticipated.
